frontend/src/components/workspaces/Templates/TemplatesTableRow/TemplatesTableRow.tsx

```diff
@@ -0,0 +1,109 @@
+import { FC } from 'react';
+import { Space, Tooltip } from 'antd';
+import Button from 'antd-button-color';
+import { TemplatesTableRowSettings } from '../TemplatesTableRowSettings';
+import {
+  DesktopOutlined,
+  CodeOutlined,
+  PlayCircleOutlined,
+} from '@ant-design/icons';
+import Badge from '../../common/Badge';
+
+export interface ITemplatesTableRowProps {
+  id: string;
+  name: string;
+  gui: boolean;
+  role: string;
+  activeInstances: number;
+  createInstance: (id: string) => void;
+  editTemplate: (id: string) => void;
+  deleteTemplate: (id: string) => void;
+}
+
+const TemplatesTableRow: FC<ITemplatesTableRowProps> = ({ ...props }) => {
+  const {
+    id,
+    name,
+    gui,
+    role,
+    activeInstances,
+    editTemplate,
+    deleteTemplate,
+  } = props;
+
+  return (
+    <>
+      <div className="w-full flex justify-between py-0">
+        <Space size={'middle'}>
+          {gui ? (
+            <DesktopOutlined style={{ color: '#1c7afd', fontSize: '24px' }} />
+          ) : (
+            <CodeOutlined style={{ color: '#1c7afd', fontSize: '24px' }} />
+          )}
+          {name}
+        </Space>
+        <div className="flex inline-flex items-center">
+          {role === 'manager' ? (
```

Comment:
The only changing part is the `TemplatesTableRowSettings`, that is visible when the user is a manager. Use conditional render only for that part, to avoid code duplication

Reply:
Changed applied [here](https://github.com/netgroup-polito/CrownLabs/pull/460/commits/77c8d7a732e466080c9e305ba922259e2ecdaa42)